src/Http/Http/test/StreamPipeWriterTests.cs

```diff
@@ -316,6 +316,30 @@ public async Task UseStreamThenPipeToWriteMultipleTimes()
             Assert.Equal(expectedString, ReadAsString());
         }
 
+        [Fact]
+        public void GetMemorySlicesCorrectly()
+        {
+            var expectedString = "abcdef";
+            var memory = Writer.GetMemory();
+
+            Encoding.ASCII.GetBytes(expectedString).CopyTo(memory);
```

Comment:
I'm not a fan of these unit tests, in my opinion they are testing the wrong thing. 

Why not check that correct data is written to the stream when calling GetMemory() - Advance() multiple times?